Support quoted attribute in traceql (#3004)
grafana/tempo: docs/sources/tempo/traceql/_index.md

{ .sla = "critical" }
```

### Quoted attribute names

> It's possible that attributes name can contain terminal characters, to search span attributes with terminal character you can use quoted attribute syntax. A quoted attirbute should be enclosed inside double quote and all characters between the quotes will be considered part of the attribute name.
> 
> For example, to find span with attribute name `attribute name with space` one can use the following query

Reviewer: ```suggestion
Attribute names can contain terminal characters. To search span attributes with terminal characters, you can use quoted attribute syntax. A quoted attribute should be enclosed inside double quotes, for example, `"example one"`. All characters between the quotes are considered part of the attribute name.

For example, to find span with attribute name `attribute name with space`, you can use the following query:
```

Author: Thanks a lot. 🙂 I have made the changes and will refer to the mentioned style guide going forward.

```
{ ."attribute name with space" = "value" }
```

> 
> Also, you can use quoted attributes syntax with non-quoted attribute syntax as well.
> 
> Fox example, below syntax is a valid traceql query.

Reviewer: Fixing some spelling and suggesting a slight wording change.

```suggestion

You can use quoted attributes syntax with non-quoted attribute syntax.

For example, the syntax below is a valid TraceQL query.
```

> Also, you can use quoted attributes syntax with non-quoted attribute syntax as well.
> 
> Fox example, below syntax is a valid traceql query.
> ```
> { span.attribute."attribute name with space" = "value" }
> ```
> 
> Note: Only `\"` and `\\` escape sequence are supported as of now.

Reviewer: ```suggestion
You can use quoted attributes syntax with non-quoted attribute syntax.

For example, below syntax is a valid TraceQL query.
```
{ span.attribute."attribute name with space" = "value" }
```

Note: Only `\"` and `\\` escape sequence are supported as of now.
```

> Note: Only `\"` and `\\` escape sequence are supported as of now.

Reviewer: This change uses the shortcode for a Note admonition. Using shortcodes helps us separate content from presentation. To learn more about the shortcode, refer to the [Writers' Toolkit](https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition). 
```suggestion
{{% admonition type="note" %}}
Currently, only `\"` and `\\` escape sequence are supported.
{{% /admonition %}}
```

{ .sla = "critical" }
```

### Quoted attribute names

> Attribute names can contain terminal characters, such as a period (`.`). To search span attributes with terminal characters, you can use quoted attribute syntax. A quoted attribute should be enclosed inside double quotes, for example, `"example one"`. All characters between the quotes are considered part of the attribute name.

Reviewer 2: nit: [Semantic line breaks](https://sembr.org/) are better for future line based diffing in the GitHub UI and result in the same HTML output.

```suggestion
Attribute names can contain terminal characters, such as a period (`.`).
To search span attributes with terminal characters, you can use quoted attribute syntax.
A quoted attribute should be enclosed inside double quotes, for example, `"example one"`.
All characters between the quotes are considered part of the attribute name.
```

> For example, to find span with attribute name `attribute name with space`, you can use the following query:
> ```
> { ."attribute name with space" = "value" }
> ```

Reviewer 2: ````suggestion
To find a span with the attribute name `attribute name with space`, use the following query:

```
{ ."attribute name with space" = "value" }
```
````

> You can use quoted attributes syntax with non-quoted attribute syntax.
> 
> For example, the syntax below is a valid TraceQL query.
> ```
> { span.attribute."attribute name with space" = "value" }
> ```

Reviewer 2: ````suggestion
You can use quoted attributes syntax with non-quoted attribute syntax, the following is a valid TraceQL query:

```
{ span.attribute."attribute name with space" = "value" }
```
````

{{% admonition type="note" %}}

> Currently, only `\"` and `\\` escape sequence are supported.

Reviewer 2: ```suggestion
Currently, only the `\"` and `\\` escape sequences are supported.
```
